sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ProjectionOverSchema.scala

```diff
@@ -41,9 +43,10 @@ case class ProjectionOverSchema(schema: StructType) {
       case a: GetArrayStructFields =>
         getProjection(a.child).map(p => (p, p.dataType)).map {
           case (projection, ArrayType(projSchema @ StructType(_), _)) =>
+            val selectedField = projSchema.find(f => resolver(f.name, a.field.name)).get
```

Comment:
It seems that we are not doing this for `struct` type. To allow this for `array of struct`, it seems that we need this for `struct` first at line 66.

Reply:
This issue can occur only in an array of structs? The code at line 66 (@dongjoon-hyun pointed out above) has the same pattern `projSchema.fieldIndex(field.name)`, so I'm worried that is can occur in other cases.

Reply:
Let me check it on.

Reply:
Oh it is fine. `ExtractValue` actually does column name resolving correctly. The difference is how `ProjectionOverSchema` treats `GetArrayStructFields` and `GetStructField` there.

That's also said we may not need to do resolving again in `ProjectionOverSchema`, as this PR currently do.  We can just use `GetArrayStructFields.ordinal` which already points to correct field in child expression.